File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if stack==[]:
                stack.append(i)
            elif i == '}':
                if stack and stack[-1]!='{':
                    return False
                stack.pop()
            elif i == ']':
                if stack and stack[-1]!='[':
                    return False
                stack.pop()
            elif i == ')':
                if stack and stack[-1]!='(':
                    return False
                stack.pop()
            else:
                stack.append(i)
        if not stack:
            return True
        return False

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for(char c: s.toCharArray()){
            if (c == '(' || c == '[' || c == '{'){
                stack.push(c);
            }

            else if (c == ')'){
                if(stack.isEmpty()) return false;
                if (!stack.peek().equals('(')) return false;
                stack.pop();
            }

            else if (c == '}'){
                if(stack.isEmpty()) return false;
                if (!stack.peek().equals('{')) return false;
                stack.pop();
            }

            else if (c == ']'){
                if(stack.isEmpty()) return false;
                if (!stack.peek().equals('[')) return false;
                stack.pop();
            }

            else{
                return false;
            }
        }
        if(stack.isEmpty()) return true;
        return false;
    }
}